lib/layers/presentation/sing_in_page/bloc/sign_in_cubit.dart

```diff
@@ -0,0 +1,158 @@
+import 'dart:async';
+import 'package:bloc/bloc.dart';
+import 'package:equatable/equatable.dart';
+import 'package:flutter/material.dart';
+import 'package:wot_statistic/generated/l10n.dart';
+import 'package:wot_statistic/common/theme/text_styles.dart';
+import 'package:wot_statistic/layers/domain/entities/user.dart';
+import 'package:wot_statistic/layers/domain/use_cases/remove_user_use_case.dart';
+import 'package:wot_statistic/layers/domain/use_cases/save_user_use_case.dart';
+import 'package:wot_statistic/layers/domain/use_cases/sing_in_use_case.dart';
+import 'package:wot_statistic/layers/domain/use_cases/subscribe_users_use_case.dart';
+import 'package:wot_statistic/layers/domain/use_cases/set_realm_pref_use_case.dart';
+import 'package:wot_statistic/layers/domain/use_cases/subscribe_realm_use_case.dart';
+
+part 'sign_in_state.dart';
+
+const notPicked = "Not Picked";
+const eu = "EU";
+
+class SingInCubit extends Cubit<SignInState> {
+  final SaveUserUseCase saveUser;
+  final SubscribeUsers subscribeUsers;
+  final SubscribeRealm subscribeRealm;
+  final SetRealmUseCase setRealm;
+  final RemoveUserUseCase removeUserUseCase;
+  final SingInUseCase signIn;
+
+  StreamSubscription? _subscriptionUsers;
+  StreamSubscription? _subscriptionRealm;
+
+  SingInCubit({
+    required this.setRealm,
+    required this.saveUser,
+    required this.subscribeUsers,
+    required this.subscribeRealm,
+    required this.removeUserUseCase,
+    required this.signIn,
+  }) : super(const SignInStateInit()) {
+    _initialize();
+  }
+
+  void _initialize() {
+    _subscriptionRealm = subscribeRealm.execute().listen((realm) {
+      if (realm == notPicked) {
+        setNewRealm(eu); // as default EU realm
```

Comment:
please remove comment and rename variable to default_region or something similar. 
Also will be great, if list of regions and default one will be in single class